src/new-components/buttons/button.tsx

```diff
@@ -25,29 +27,32 @@ const StyledButton = styled.button`
 `;
 
 const StyledIcon = styled(Icon)`
-  padding-right: 10px;
-  // pull the fixed size icon closer to the text
-  margin-left: -4px;
-  /**
-    This is a magic number to match the vertical middle of the button
-    with the icon which has a fixed bounding box which won't allow us to align
-    it based on the actual shape of the icon.
-   */
-  vertical-align: -18%;
+  margin-right: 10px;
 `;
 
 const ButtonText = styled.span`
+  ${TextStyles.toplineR};
+  line-height: 24px;
   padding-left: 16px;
-  padding-right: 16px;
+  padding-right: 6px;
 `;
 
-type RegularButtonProps = React.ComponentPropsWithoutRef<'button'>;
-
-export const Button = (props: RegularButtonProps) => {
-  return (
-    <StyledButton {...props}>
-      <ButtonText>{props.children}</ButtonText>
-      <StyledIcon show={'chevron_right'} size={IconSize.NORMAL} />
-    </StyledButton>
-  );
+type RegularButtonProps = React.ComponentPropsWithoutRef<'button'> & {
+  to?: string;
 };
+
+export const Button = (props: RegularButtonProps) => (
+  <>
+    {props.to ? (
+      <StyledButton to={props.to}>
+        <ButtonText>{props.children}</ButtonText>
+        <StyledIcon show={'chevron_right'} size={IconSize.NORMAL} />
+      </StyledButton>
+    ) : (
+      <StyledButton as="button" {...props}>
+        <ButtonText>{props.children}</ButtonText>
+        <StyledIcon show={'chevron_right'} size={IconSize.NORMAL} />
+      </StyledButton>
+    )}
+  </>
+);
```

Comment:
I think this would be much more readable with an explicit if-else condition. The ternary operator always makes reading code harder.

```suggestion
export const Button = (props: RegularButtonProps) => {
  if (props.to) {
    return (
      <StyledButton to={props.to}>
        <ButtonText>{props.children}</ButtonText>
        <StyledIcon show={'chevron_right'} size={IconSize.NORMAL}/>
      </StyledButton>
    )
  } else {
    return (
      <StyledButton as="button" {...props}>
        <ButtonText>{props.children}</ButtonText>
        <StyledIcon show={'chevron_right'} size={IconSize.NORMAL}/>
      </StyledButton>
    )
  }
};

```